packages/xo-web/src/common/xo/index.js

```diff
@@ -1108,11 +1135,38 @@ export const pauseVms = vms =>
 
 export const recoveryStartVm = vm => _call('vm.recoveryStart', { id: resolveId(vm) })
 
-export const restartVm = (vm, force = false) =>
-  confirm({
-    title: _('restartVmModalTitle'),
-    body: _('restartVmModalMessage', { name: vm.name_label }),
-  }).then(() => _call('vm.restart', { id: resolveId(vm), force }), noop)
+export const restartVm = async (vm, force = false) => {
+  try {
+    await confirm({
+      title: _('restartVmModalTitle'),
+      body: _('restartVmModalMessage', { name: vm.name_label }),
+    })
+  } catch {
+    return
+  }
+
+  try {
+    if (
+      !vm.pvDriversDetected &&
+      !force &&
+      (await chooseAction({
+        title: _('vmHasNoTools'),
+        body: _('vmHasNoToolsMessage'),
+        icon: 'alarm',
+        buttons: [
+          { label: _('tryReboot'), value: 'clean', btnStyle: 'success' },
+          { label: _('forceRebootVmLabel'), value: 'force', btnStyle: 'danger' },
+        ],
+      })) === 'force'
+    ) {
+      force = true
+    }
+  } catch {
+    return
+  }
```

Comment:
Why don't we simply try a clean reboot and if that does not work and the error is that it's unavailable, we ask if the user want to force?

Reply:
That's in the initial spec by @stormi

Reply:
What is in the initial spec?

There were at least 3 proposals on this issue :upside_down_face:

Reply:
Quoting:
> if the VM tools are not installed, not even try to do a soft reboot/shutdown and ask user whether they want to do a hard reboot/shutdown instead

Reply:
This is proposal 2 on 3 (4 if you count mine).

Reply:
We should ask @stormi why he'd prefer not even waiting for the error.

Reply:
I confirm I implemented proposal #2, which is why I did it like that. But it can be changed, of course.

I find the user experience slightly better like this but I agree that relying on a detection isn't ideal. Waiting for @stormi's opinion.

Reply:
FYI he's back on the 28th :)

Reply:
Why is the user experience better?

Reply:
Ok, let's be safe and react to the error then. We can always change it in XO 6 based on feedback.

Reply:
I did agree to "react to the error" when Marc² asked me, though the test before approach would have been fine to me too. I can give feedback on the current implementation if there's a test instance available.